apps/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -31,31 +34,66 @@ export class PackageChangeAnalyzer {
     this._git = new Git(this._rushConfiguration);
   }
 
-  public async getPackageDeps(
+  /**
+   * Try to get a list of the specified project's dependencies and their hashes.
+   *
+   * @remarks
+   * If the data can't be generated (i.e. - if Git is not present), this throws an Error.
+   * If the project name is invalid, this returns undefined.
+   */
+  public async getProjectDependenciesAsync(
```

Comment:
"Dependencies" sounds like package.json dependencies.  Is there a better term?

Reply:
"FileHashes?"

Reply:
From an API design standpoint, the `Map<string, string>` is the problem.  It is the centerpiece of this optimization, but it is undocumented and does not even have a name.   This design flaw is inherited from `@rushstack/package-deps-hash` which never got reviewed.

🤔 Since `@rushstack/package-deps-hash` is [pretty much only used by Rush](https://www.npmjs.com/browse/depended/@rushstack/package-deps-hash) we could probably revamp it pretty easily.  

Looking at the implementation, it is entirely based around Git and has no ties to the concepts of a "package" or "dependencies".  It's more like *"you give me a folder, and I'll give you the hashes for Git-tracked files under that folder."*  

Maybe we could deprecate `@rushstack/package-deps-hash` entirely and replace it with a `@rushstack/git-client` that can eventually consolidate all the APIs for interacting with Git.

For the `Map<string, string>` problem, I'd suggest to make a class called something like `GitHashMap` that implements its own comparisons and knows how to serialize/deserialize itself to JSON.

Then this API would become:

```ts
ProjectChangeAnalyzer.getGitHashMapAsync(projectName: string, terminal: Terminal): Promise<GitHashMap>`
```

Reply:
The tool I'm currently working on doesn't need this API, so what do you think about marking it as `@internal` for now?

Reply:
👍 that is a good way to get this PR merged